presentation/components/src/presentation-components/tree/controlled/TreeHooks.ts

```diff
@@ -55,284 +55,284 @@ export interface PresentationTreeNodeLoaderProps extends PresentationTreeDataPro
-  useResumeNodeLoading(modelSource, nodeLoader);
   return { nodeLoader, onItemsRendered };
 }
 
-/** Starts loading children for nodes that are marked as loading, each time arguments change. */
-function useResumeNodeLoading(
-  modelSource: TreeModelSource,
-  nodeLoader: PagedTreeNodeLoader<IPresentationTreeDataProvider>,
-): void {
-  useEffect(
-    () => {
-      let subscriptions: Subscription | undefined;
-      for (const node of modelSource.getModel().iterateTreeModelNodes()) {
-        if (!node.isLoading) {
-          continue;
-        }
-
-        if (subscriptions === undefined) {
-          subscriptions = nodeLoader.loadNode(node, 0).subscribe();
-        } else {
-          subscriptions.add(nodeLoader.loadNode(node, 0).subscribe());
-        }
-      }
-
-      return () => subscriptions?.unsubscribe();
-    },
-    [modelSource, nodeLoader],
-  );
-}
-
-interface ModelSourceUpdateProps {
-  enable?: boolean;
+interface TreeNodeLoaderState {
   modelSource: TreeModelSource;
-  rerenderWithTreeModel: (treeModel?: MutableTreeModel) => void;
+  rulesetRegistration: RulesetRegistrationHelper;
   dataProvider: IPresentationTreeDataProvider;
-  treeNodeItemCreationProps: CreateTreeNodeItemProps;
 }
 
-function useModelSourceUpdateOnIModelHierarchyUpdate(props: ModelSourceUpdateProps) {
-  const { modelSource, dataProvider, rerenderWithTreeModel } = props;
-  useExpandedNodesTracking({ modelSource, dataProvider, enableNodesTracking: !!props.enable });
-  const renderedItems = useRef<RenderedItemsRange | undefined>(undefined);
-  const onItemsRendered = useCallback((items: RenderedItemsRange) => {
-    renderedItems.current = items;
-  }, []);
-
-  const onIModelHierarchyChanged = useCallback(
-    async (args: IModelHierarchyChangeEventArgs) => {
-      if (args.rulesetId === dataProvider.rulesetId && args.imodelKey === dataProvider.imodel.key) {
-        if (args.updateInfo === UPDATE_FULL) {
-          rerenderWithTreeModel(undefined);
-        } else {
-          await updateModelSourceAfterIModelChange(modelSource, rerenderWithTreeModel, args.updateInfo, dataProvider, props.treeNodeItemCreationProps, renderedItems.current);
-        }
-      }
-    },
-    [modelSource, dataProvider, rerenderWithTreeModel, props.treeNodeItemCreationProps],
-  );
-  useEffect(
-    () => {
-      return props.enable
-        ? Presentation.presentation.onIModelHierarchyChanged.addListener(onIModelHierarchyChanged)
-        : undefined;
-    },
-    [onIModelHierarchyChanged, props.enable],
-  );
-
-  return onItemsRendered;
-}
+/**
+ * Resets state to `initialValue` when dependencies change. Avoid using in new places because this hook is only intended
+ * for use in poorly designed custom hooks.
+ */
+function useResettableState<T>(initialValue: () => T, dependencies: unknown[]): [T, React.Dispatch<React.SetStateAction<T>>] {
+  const stateRef = React.useRef<T>() as React.MutableRefObject<T>;
+  const propsChangedRef = React.useRef(true);

```

Comment:
Looks like this is not used anymore

Reply:
Thanks, removed unused variable in 00a327200301ce09fefd3c33fa831171a933e8cc.

Reply:
Thanks, removed unused variable in 00a327200301ce09fefd3c33fa831171a933e8cc.